framework/src/util/SharedLibrary.cpp

```diff
@@ -88,35 +94,51 @@ SharedLibrary::SharedLibrary(const std::string& absoluteFilePath)
 
 SharedLibrary::~SharedLibrary() = default;
 
-SharedLibrary& SharedLibrary::operator=(const SharedLibrary&) = default;
+SharedLibrary& SharedLibrary::operator=(const SharedLibrary& other)
+{
+  SharedLibraryPrivate* slp = new SharedLibraryPrivate;
```

Comment:
`d.reset(new SharedLibraryPrivate(*other.d))` should be enough, plus the `m_Handle` check and library loading as in the copy constructor.

Reply:
For the reason above about keeping the handle as a std::shared_ptr, I'm not implementing the opening part of your comment unless you decide that you truly want it to be a raw pointer.

Reply:
Yes, but due to my comment above to your other post, unless you absolutely want to keep the handle as a raw pointer, then I won't be doing the extra Load() call in this PR.